sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -464,6 +464,19 @@ class Dataset[T] private[sql](
    */
   def as[U : Encoder]: Dataset[U] = Dataset[U](sparkSession, logicalPlan)
 
+  /**
+   * Returns a new DataFrame where each row is reconciled to match the specified schema. Spark will:
```

Comment:
@cloud-fan Could we also mention the behavior regarding nullability in the comments.